src/core/execute.c

```diff
@@ -3216,11 +3216,18 @@ static int exec_child(
                 }
         }
 
-        if (context->cpu_set.set)
-                if (sched_setaffinity(0, context->cpu_set.allocated, context->cpu_set.set) < 0) {
+        if (context->cpu_set.set || context->numa_policy.derive_cpu_affinity_from_numa_mask) {
+                r = exec_context_cpu_affinity_from_numa((ExecContext *) context);
```

Comment:
hu? nooo. we don't cast away `const`, because `const` means constant, and we don't modify constant stufff...

make the function return the CPUset to use, and then apply that.

Reply:
Will do. Though, the resulting code will be a bit more complicated and slower.

I thought it would be acceptable to cast away const since this is running in child after fork() and original ExecContext is not changed.

Reply:
We can spare those cycles... it's in a NUMA-only codepath after all... And even if it wasn't, it still would be OK.

And no, let's not start with making stuff writable that is marked as immutable. So far ExecContext is entirely immutable after loading, and I think it should stay that way, it makes things easier to grasp.